src/main/java/org/broadinstitute/hellbender/engine/filters/OverclippedReadFilter.java

```diff
@@ -14,9 +14,9 @@
  *     This filter is intended to filter out reads that are potentially from foreign organisms.
  *     From experience with sequencing of human DNA we have found cases of contamination by bacterial
  *     organisms; the symptoms of such contamination are a class of reads with only a small number
- *     of aligned bases and additionally many soft-clipped bases.  This filter is intended
- *     to remove such reads. Consecutive soft-clipped blocks are treated as a single block
- * </p>
+ *     of aligned bases and additionally many soft-clipped bases. This filter is intended
+ *     to remove such reads.
```

Comment:
I'm not familiar with this filter, but as far as I understood, this filter does two things:

1. Filter out reads where the number of bases without soft-clips (M, I, X, and = CIGAR operators) is lower than a threshold.
2. If it have more than 2 soft-clip blocks (or 1, if `--doNotRequireSoftclipsOnBothEnds` is provided), then it is also "overclipped".

Thus, hard-clipped bases are counted to some extend: if the read is something like `5S29M1S65H`, it will be discarded because it is too short (due to hard-clips).